training/src/main/scala/com/airbnb/aerosolve/training/Evaluation.scala

```diff
@@ -197,6 +330,16 @@ object Evaluation {
     metrics.append(("!TRAIN_PRECISION", if (ttp > 0.0) ttp / (ttp + tfp) else 0.0))
     metrics.append(("!TRAIN_F1", 2 * ttp / (2 * ttp + tfn + tfp)))
 
+    if (results != null) {
```

Comment:
maybe it is better to report it with metrics.append, i.e. metricsAppend(metrics, results, name, value)